libvast/src/system/tracker.cpp

```diff
@@ -28,6 +28,86 @@ using namespace caf;
 namespace vast {
 namespace system {
 
+namespace {
+
+struct terminator_state {
+  terminator_state(event_based_actor* selfptr) : self(selfptr) {
+    // nop
+  }
+
+  caf::error reason;
+  actor parent;
+  component_state_map components;
+  std::vector<std::string> victim_stack;
+  size_t pending_down_messages = 0;
+  event_based_actor* self;
+
+  void init(caf::error reason, caf::actor parent,
+            component_state_map components,
+            std::vector<std::string> victim_stack) {
+    this->reason = std::move(reason);
+    this->parent = std::move(parent);
+    this->components = std::move(components);
+    this->victim_stack = std::move(victim_stack);
+  }
+
+  template <class Label>
+  void kill(const caf::actor& whom, const Label& label) {
```

Comment:
This should probably be called `who`, since `label` is the indirect object and the actor to kill the direct object.